### PR TITLE
New travis config, rasterio 1.0 compat

### LINKED ISSUE #14: wrond dataset of mbtiles and png/jpg files
I install rio-mbtiles on Ubuntu 14.04.3 and try create mbtiles from my raster and raster in tests/data/ folder:  
rio mbtiles RGB.byte.tif -o raster.mbtiles -j 4 --format PNG --image-dump /home/box/rio-mbtiles/tests/data
mbtiles and image tiles successfully created but image tiles (png and jpg) don't opened in standart image files viewer. Also mbtiles file don't opened in QGIS (opened as tables (tiles and metadata)) or another mbtiles viewer. Example output data available on link https://www.dropbox.com/s/kfg2pywe9q21adx/rio-mbtiles_data.zip?dl=0
How I can get valid mbtiles?

### LINKED ISSUE #17: need way to force overwrite, rasterio 1.0 compat
Recent Rasterio versions require explicit flags to overwrite existing files.

Two problems here:
1. Our test suite tries to overwrite the test fixtures. They should probably just be written to temp space, right?
2. We have no way to specify `--force-overwrite` at the CLI should we want this behavior. 

In `cli.py` we use `from rasterio.rio.helpers import resolve_inout` which takes a ` force_overwrite` keyword argument. We could pass `True` and keep the old behavior if needed.

### LINKED ISSUE #19: travis fails to build Rasterio
`rasterio/_base.c:287:22: fatal error: cpl_conv.h: No such file or directory` - looks like no python-dev. In fact, looks like this repo is set up to use old linux wheels.

@sgillies maybe this would be a good place to employ the manylinux wheels for 1.0a1?

### PR DESCRIPTION
**Pull request by sgillies**
I originally intended that this would be about using manylinux wheels, but those are blocked by the issue noted in https://github.com/mapbox/rasterio/issues/693#issuecomment-249584437.

Building Rasterio from an sdist is only taking 70 seconds now, which isn't super fast but not a plugin killer.

Resolves #19 
Resolves #14 
Resolves #17

### CODE REVIEW

**Comment by sgillies** [comment]
@perrygeo ready for review and merge.

File: .travis.yml [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-    - $HOME/.pip-cache/
-    - $HOME/wheelhouse
-before_install:
-  - "./.travis_riodeps.sh"
+    - $HOME/.cache/pip/
 install:
-  - "pip install -U pip"
-  - "pip install --use-wheel --find-links=$HOME/wheelhouse -e .[test] --cache-dir $HOME/.pip-cache"
+  - pip install -U pip
+  - pip install --pre --no-binary rasterio rasterio==1.0a1
```
Comment (perrygeo) [inline .travis.yml]: should we use `'rasterio>=1.0a1'` to keep it up to date?

**Comment by perrygeo** [comment]
Also, do you mind if I "Squash and Merge" this one?

File: .travis.yml (hunk 1 above)
Comment (sgillies) [inline .travis.yml]: I'll do that and remove the `--no-binary` option right before 1.3.0.

**Comment by sgillies** [comment]
@perrygeo sure, squash and merge when the light is green.

**Pull request merged**